File: ui/weather/temperature_label.py
from gi.repository import Gtk


class TemperatureLabel(Gtk.Label):
    def __init__(self):
        super().__init__()
        self.set_label("??")

    def set_temperature(self, temperature):
        self.set_label(f"{temperature:>5.1f}℃")

File: ui/src/weather/temperature_label.c
#include "ui/include/weather/temperature_label.h"

GtkWidget *temperature_label_new() { return gtk_label_new("??"); }

void temperature_label_refresh(GtkWidget *label, float temperature) {
  char buffer[100];
  sprintf(buffer, "%5.1f℃", temperature);
  gtk_label_set_label(GTK_LABEL(label), buffer);
}
